src/lyk.c

```diff
@@ -6,75 +6,136 @@
 #include "ldebug.h"
 #include <stdlib.h>
 #include <yk.h>
-
-#define _DEFAULT_SOURCE /* for reallocarray() */
-
+#include "stdio.h"
 /*
  *  Function prototypes in Lua are loaded through two methods:
  *    1.  They are loaded from text source via the `luaY_parser`.
  *    2.  They can also be loaded from binary representation using `luaU_undump`, where
  *        prototypes are previously dumped and saved.
- * 
- *  Yk tracing locations are allocated using both of these methods 
+ *
+ *  Yk tracing locations are allocated using both of these methods
  *  using `yk_set_location` and `yk_set_locations` respectively.
-*/
+ */
+
+static int LYK_VERBOSE = -1;
+
+int is_verbose() {
+  if (LYK_VERBOSE == -1) {
+    LYK_VERBOSE = getenv("LYK_VERBOSE") != NULL;
+  }
+  return LYK_VERBOSE;
+}
+
+void print_proto_info(char *msg, Proto *f) {
```

Comment:
I think I hear Laurie's footsteps approaching...

I'm OK with leaving this debugging code in for now, but once we have everything solved, we should remove it again.

Reply:
Unless there's a convincing reason, we should not include it, because we'll never remember to remove it.

Reply:
Also all of the debugging stuff should not be compiled-in for non-devug builds, since they will impact performance.

See `NDEBUG` in the `assert` manual page for more.

Reply:
Here's my argument:

I've included these logs cause I have been stuck in this loop since I started working on yklua:
1. Add prints in yk related code
2. debug + find the issue
3. Remove print and open PR
4. go to 1

I think I've done it more than 10 times by now, so I would argue that we should keep it until we have a stable Yklua interpreter.

I understand the concern of maybe forgetting to remove it when we're done. But on the other hand, it will allow me to iterate faster.

I'm also OK with removing it from this PR change and keeping the code clean.

Reply:
I'll leave it to your discretion as to whether we keep it in or not, but if we do keep it in, we'd need to `#ifdef` every single instance so that we know what to remove in the future.

Reply:
I feel your pain @Pavel-Durov.

Since Laurie is letting you keep it in, I'd jump on that. Just make sure it's all guarded `#ifndef NDEBUG`.

Reply:
done :point_right: 50489548aff4a783800f226322f6199887e21292

Reply:
Oops, I wasn't clear. We need to use a flag of our own devising so that we can later differentiate "we added this block" from "Lua uses `NDEBUG` for other purposes". [Admittedly I only see two `NDEBUG`s in Lua, but it's greater than zero.]

Reply:
ok np

Reply:
done :point_right: 632330f91632ecc89e641dfcc7b944149d5d3572